src/components/ebay-icon/stamper/index.js

```diff
@@ -0,0 +1,87 @@
+const fs = require('fs');
+const template = require('./template.marko');
+
+const pageCache = {};
+const markupCache = {};
+const markoFileReg = /\.marko(?:\.js)?$/;
+
+/**
+ * Traverse the require tree, starting at a specified rootNode
+ * @param {*} rootNode: starting node for traversal
+ * @param {*} fn: function to execute on every child
+ */
+function traverse(rootNode, fn) {
+    const visited = {};
+    const queue = [rootNode];
+
+    while (queue.length > 0) {
+        const currentNode = queue.shift();
+        if (!currentNode || visited[currentNode.id]) {
+            continue;
+        }
+        visited[currentNode.id] = 1;
+        fn(currentNode);
+
+        if (currentNode.children && currentNode.children.length) {
+            // unshift for dfs, push for bfs
+            currentNode.children.forEach(child => queue.unshift(child));
+        }
+    }
+}
+
+/**
+ * Use traversal to find icons in the require cache for this page
+ * @param {*} pageTemplateNode: first require cache module of the page
+ * @param {*} pageTemplateId: require cache ID of the page module
+ */
+function findIcons(out) {
+    const pageTemplatePath = out && out.global && out.global.pageTemplate && out.global.pageTemplate.path || '';
+    const pageTemplatePathMarko = pageTemplatePath.replace('.marko.js', '.marko');
+    const pageTemplateNode = require.cache[pageTemplatePath] || require.cache[pageTemplatePathMarko];
+
+    if (!pageTemplateNode) {
+        return;
+    }
+
+    const pageTemplateId = pageTemplateNode.id;
+    let icons = pageCache[pageTemplateId];
+
+    // only traverse icon paths once per page template
+    if (icons) {
+        return icons;
+    }
+
+    icons = pageTemplateNode._ebay_icons || [];
+    traverse(pageTemplateNode, currentNode => {
+        if (markoFileReg.test(currentNode.id)) {
+            const foundIcons = currentNode.exports && currentNode.exports._ebay_icons;
+
+            if (foundIcons) {
+                icons.push(...foundIcons);
+            }
+        }
+    });
+
+    // remove duplicates
+    pageCache[pageTemplateId] = icons.filter((value, index) => icons.indexOf(value) === index);
```

Comment:
Micro optimization would be to use a `Set` here for the `icons` and then we don't need to remove duplicates ourselves.

Reply:
Maybe we should defer this until we set up the babel runtime transform https://github.com/eBay/ebayui-core/issues/156

Reply:
@DylanPiercey I looked at using `Set`, but it seems like it might not be worth it. We're doing other array stuff like `icons.map()`, and using `Set` would mean conversions like `Array.from(icons.values()).map()`. So probably simplest to use array across the board.

Reply:
Alrighty, lets hold off on this then.